coffea/processor/executor.py

```diff
@@ -259,7 +245,33 @@ def _futures_handler(futures, timeout):
             _cancel(futures.pop())
 
 
-def work_queue_executor(items, function, accumulator, **kwargs):
+@dataclass
+class ExecutorBase:
+    # shared by all executors
+    status: bool = True
+    unit: str = "items"
+    desc: str = "Processing"
+    compression: Optional[int] = 1
+    function_name: Optional[str] = None
+
+    def __call__(
+        self,
+        items: Sequence,
```

Comment:
`items`  could also be a generator or iterator, at least for WorkQueueExecutor. However, I'm not sure if that should be specified here, or just for Work Queue.  I think that that Sequence does not cover generators, as they lack __len__.

Reply:
yep, it should be the `Iterable` protocol, which implements `__iter__`.